lib/protocol/protocol.go

```diff
@@ -867,26 +870,29 @@ func (c *rawConnection) shouldCompressMessage(msg message) bool {
 // BEP message is sent before terminating the actual connection. The error
 // argument specifies the reason for closing the connection.
 func (c *rawConnection) Close(err error) {
-	c.sendCloseOnce.Do(func() {
-		done := make(chan struct{})
-		timeout := time.NewTimer(CloseTimeout)
-		select {
-		case c.closeBox <- asyncMessage{&Close{err.Error()}, done}:
+	// Close might be called from a method that is called from within
+	// dispatcherLoop, resulting in a deadlock.
+	// The sending above must happen before spawning the routine, to prevent
```

Comment:
I don't understand this comment. Which sending above?

Reply:
Thanks so much - I just didn't read that comment at all and it is explaining why I can't do what I did in this PR before. It's nice that I write comments (often after some nudging), but I really also need to read them...